### DISCUSSION TITLE
Is this correct to calculate the surface binding energy by cp2k?

### DISCUSSION THREAD

**Discussion by lemon1044** [cp2k/cp2k#3753] [Q&A]
Through cp2k calculation, the binding energy of glucose molecules adsorbed on the substrate
Because I don’t know how glucose is adsorbed on the substrate
I want to fix the substrate material and optimize the geometry to get the minimum energy A of the total system (glucose adsorbed on the substrate)

Finally, the adsorption binding energy E=A-E(glucose)-E(substrate)
Is there anything wrong with this idea (I am a beginner in cp2k and I am not sure whether the binding energy should be calculated in this way. If you can recommend a similar case, I would be very grateful)
The following is my input file

(This is the result of my file after running it.

 SCF run NOT converged. To continue the calculation regardless,   *
*    |             please set the keyword IGNORE_CONVERGENCE_FAILURE.         *

 It may be that the convergence value is too small. But I mainly want to ask whether the idea of ​​my calculation is correct.)

#Generated by Multiwfn (http://sobereva.com/multiwfn)
&GLOBAL
  PROJECT basis-glu-geo
  PRINT_LEVEL LOW
  RUN_TYPE GEO_OPT
&END GLOBAL

&FORCE_EVAL
  METHOD Quickstep
  &SUBSYS
    &CELL
      A    20.63360000     0.00000000     0.00000000
      B     0.00000000    22.42308000     0.00000000
      C     0.00000000     0.00000000    25.16297000
      PERIODIC XYZ #Direction(s) of applied PBC (geometry aspect)
    &END CELL
    &COORD
      Ti         10.53378000    0.01215000    3.73742000
      Ti         10.49260000    6.03917000    3.69370000
      Ti          5.25049000   -0.24599000    3.76264000
      Ti          5.19184000    6.05541000    3.72545000
      Ti         -0.09982000    3.06172000    3.78350000
      Ti          0.43065000    9.68942000    3.69653000
      Ti          5.19089000    2.97813000    3.70302000
      Ti          5.15434000    9.15675000    3.70781000
      Ti          2.59715000    4.49177000    3.75684000
      Ti          2.46949000   10.30866000    3.89692000
      Ti          7.82272000    4.48938000    3.66011000
      Ti          7.79170000   10.81198000    3.68392000
      Ti          2.59620000    1.40618000    3.74218000
      Ti          2.50542000    7.46446000    3.85689000
      Ti          7.74379000    1.39237000    3.67745000
      Ti          7.84505000    7.60120000    3.63781000
      Ti          1.80101000   12.11458000    1.10108000
      Ti          1.65298000    6.07006000    1.26501000
      Ti          7.01959000   12.17709000    1.04621000
      Ti          7.28946000    6.08979000    1.01426000
      Ti          1.70702000    2.99977000    1.19376000
      Ti          1.68912000    9.06357000    1.12568000
      Ti          7.08758000    3.04091000    1.08576000
      Ti          6.93318000    9.09511000    1.15006000
      Ti          4.67454000    4.48319000    1.03453000
      Ti          4.54679000   10.76515000    1.12515000
      Ti          9.88879000    4.79893000    0.82100000
      Ti          9.87149000   10.19143000    1.21885000
      Ti          4.21653000    1.41240000    1.07103000
      Ti          4.66007000    7.58132000    1.02650000
      Ti          9.61555000    1.85140000    1.18997000
      Ti          9.83011000    7.31393000    1.02789000
      Ti          0.56219000    1.35134000    6.24449000
      Ti          0.42995000    7.26139000    6.60454000
      Ti          5.96171000    1.45459000    6.28490000
      Ti          5.73503000    7.58627000    6.39420000
      Ti          0.53095000    4.72694000    6.21810000
      Ti          0.77287000   10.69977000    6.35388000
      Ti          5.70717000    4.47955000    6.42531000
      Ti          5.94602000   10.69281000    6.45379000
      Ti          3.26218000    6.09639000    6.44724000
      Ti          3.34021000   -0.11570000    6.51570000
      Ti          8.68865000    6.04511000    6.21696000
      Ti          8.64814000   -0.07028000    6.29363000
      Ti          3.09534000    3.00038000    6.40875000
      Ti          3.42674000    9.17960000    6.33499000
      Ti          8.74063000    3.00727000    6.17929000
      Ti          8.69362000    9.07747000    6.17189000
      C           0.85597000    1.58313000    2.54491000
      C           0.89655000    7.73755000    2.55394000
      C           5.87754000    1.40529000    2.59243000
      C           5.93434000    7.58134000    2.68304000
      C           0.93269000    4.45452000    2.49774000
      C           1.01352000   10.56600000    2.22031000
      C           5.94869000    4.53616000    2.67315000
      C           6.05249000   10.45004000    2.48586000
      C           3.31667000    6.05095000    2.64703000
      C           3.59259000    0.00909000    2.55469000
      C           8.61043000    6.01796000    2.63217000
      C           8.89024000    0.16924000    2.39711000
      C           3.33800000    2.99164000    2.62681000
      C           3.23174000    9.08365000    2.57775000
      C           8.43677000    2.84400000    2.55656000
      C           8.48083000    9.25182000    2.47733000
      C           1.81449000    0.05257000    5.07422000
      C           1.87251000    5.85708000    4.99399000
      C           6.91529000    0.08870000    5.04292000
      C           7.07195000    6.05106000    4.78299000
      C           1.80407000    2.96598000    4.78084000
      C           1.67186000    8.68207000    5.27675000
      C           7.04952000    2.98365000    4.78984000
      C           7.06680000    9.13575000    4.75049000
      C           4.44483000    4.51439000    4.76920000
      C           4.28671000   10.65713000    4.97033000
      C           9.46839000    4.58270000    4.92924000
      C           9.49761000   10.60360000    4.94638000
      C           4.45829000    1.42626000    4.78318000
      C           4.42516000    7.57449000    4.77168000
      C           9.38536000    1.34535000    4.96016000
      C           9.47648000    7.50053000    4.93869000
      O          -0.01243000   -0.02138000    0.22166000
      O           0.04038000    6.05578000    0.15754000
      O           5.02432000    0.00177000   -0.00173000
      O           5.08489000    6.03039000   -0.05689000
      O           0.07551000    3.00534000    0.10867000
      O           0.04495000    9.04497000    0.06146000
      O           5.21847000    2.82245000    0.17880000
      O           4.93817000    9.20231000   -0.03185000
      O           2.41601000    4.55598000    0.16149000
      O           2.44724000   10.53825000    0.03814000
      O           7.67187000    4.51649000   -0.01709000
      O           7.69757000   10.57067000    0.12531000
      O           2.55249000    1.57346000    0.06602000
      O           2.41505000    7.52089000    0.13534000
      O           7.92531000    1.57954000    0.19123000
      O           7.73373000    7.70851000    0.02812000
      O          10.34433000   -0.03068000    7.28154000
      O          10.31646000    6.04488000    7.31193000
      O           5.16230000   -0.04795000    7.20047000
      O           5.30087000    6.05678000    7.47023000
      O          10.35863000    3.02054000    7.28032000
      O          10.30099000    9.08918000    7.30946000
      O           5.16388000    2.85036000    7.35529000
      O           5.33611000    9.15089000    7.47066000
      O           2.72522000    4.55991000    7.49916000
      O           2.48293000   10.56726000    7.30225000
      O           7.98476000    4.51896000    7.27372000
      O           7.81639000   10.54191000    7.16680000
      O           2.66959000    1.43442000    7.47232000
      O           2.75493000    7.71952000    7.44270000
      O           8.02484000    1.54381000    7.30472000
      O           7.94902000    7.56817000    7.27634000
      H           5.66004000    1.62122000   10.42852000
      H           5.24664000    5.43919000   13.35446000
      H           6.66541000    5.28466000   15.10608000
      H           5.33627000    3.76544000   11.69796000
      H           7.03741000    1.95262000   12.21711000
      O           6.60485000    1.84161000   10.22592000
      C           6.34172000    5.61444000   13.22170000
      O           6.88830000    6.00003000   14.45506000
      C           7.11957000    2.57834000   11.30647000
      C           6.43039000    3.94274000   11.57368000
      O           6.99462000    4.43628000   12.78029000
      H           4.93153000    6.16488000   10.91023000
      O           5.91025000    7.90788000   12.65988000
      C           6.57273000    6.75074000   12.21495000
      H           6.34545000    3.68126000    9.05830000
      C           6.03681000    6.31892000   10.84492000
      H           8.19873000    2.75912000   11.11699000
      C           6.66306000    4.97786000   10.44302000
      O           6.04520000    4.58406000    9.26363000
      H           6.41165000    8.68906000   12.30915000
      H           7.67312000    6.92396000   12.15221000
      H           7.75723000    5.11771000   10.28082000
      H           5.60430000    7.95107000    9.87744000
      O           6.33811000    7.28356000    9.86823000
    &END COORD
    &KIND Ti
      ELEMENT Ti
      BASIS_SET DZVP-MOLOPT-SR-GTH-q12
      POTENTIAL GTH-PBE
    &END KIND
    &KIND C
      ELEMENT C
      BASIS_SET DZVP-MOLOPT-SR-GTH-q4
      POTENTIAL GTH-PBE
    &END KIND
    &KIND O
      ELEMENT O
      BASIS_SET DZVP-MOLOPT-SR-GTH-q6
      POTENTIAL GTH-PBE
    &END KIND
    &KIND H
      ELEMENT H
      BASIS_SET DZVP-MOLOPT-SR-GTH-q1
      POTENTIAL GTH-PBE
    &END KIND
  &END SUBSYS

  &DFT
    BASIS_SET_FILE_NAME  BASIS_MOLOPT
    POTENTIAL_FILE_NAME  POTENTIAL
#   WFN_RESTART_FILE_NAME Ti3C2O2-glu-geo-RESTART.wfn
    CHARGE    0 #Net charge
    MULTIPLICITY    1 #Spin multiplicity
    &QS
      EPS_DEFAULT 1.0E-12 #Set all EPS_xxx to values such that the energy will be correct up to this value
    &END QS
    &POISSON
      PERIODIC XYZ #Direction(s) of PBC for calculating electrostatics
      PSOLVER PERIODIC #The way to solve Poisson equation
    &END POISSON
    &XC
      &XC_FUNCTIONAL PBE
      &END XC_FUNCTIONAL
      &VDW_POTENTIAL
        POTENTIAL_TYPE PAIR_POTENTIAL
        &PAIR_POTENTIAL
          PARAMETER_FILE_NAME dftd3.dat
          TYPE DFTD3(BJ)
          REFERENCE_FUNCTIONAL PBE
          #CALCULATE_C9_TERM T #Calculate C9-related three-body term, more accurate for large system
        &END PAIR_POTENTIAL
      &END VDW_POTENTIAL
    &END XC
    &MGRID
      CUTOFF  400
      REL_CUTOFF  55
    &END MGRID
    &SCF
      MAX_SCF 128
      EPS_SCF 1.0E-06 #Convergence threshold of density matrix of inner SCF
#     SCF_GUESS RESTART #Use wavefunction from WFN_RESTART_FILE_NAME file as initial guess
#     IGNORE_CONVERGENCE_FAILURE #Continue calculation even if SCF not converged, works for version >= 2024.1
      &DIAGONALIZATION
        ALGORITHM STANDARD #Algorithm for diagonalization
      &END DIAGONALIZATION
      &MIXING #How to mix old and new density matrices
        METHOD BROYDEN_MIXING #PULAY_MIXING is also a good alternative
        ALPHA 0.4 #Default. Mixing 40% of new density matrix with the old one
        NBROYDEN 8 #Default is 4. Number of previous steps stored for the actual mixing scheme
      &END MIXING
      &SMEAR
        METHOD FERMI_DIRAC
        ELECTRONIC_TEMPERATURE 300 #Electronic temperature of Fermi-Dirac smearing in K
      &END SMEAR
      ADDED_MOS    68 #Number of virtual MOs to solve
      &PRINT
        &RESTART #Note: Use "&RESTART OFF" can prevent generating .wfn file
          BACKUP_COPIES 0 #Maximum number of backup copies of wfn file. 0 means never
        &END RESTART
      &END PRINT
    &END SCF
  &END DFT
&END FORCE_EVAL

&MOTION
  &GEO_OPT
    TYPE MINIMIZATION #Search for minimum
    KEEP_SPACE_GROUP F #If T, then space group will be detected and preserved
    OPTIMIZER BFGS #Can also be CG (more robust for difficult cases) or LBFGS
    &BFGS
      TRUST_RADIUS 0.2 #Trust radius (maximum stepsize) in Angstrom
#     RESTART_HESSIAN T #If read initial Hessian, uncomment this line and specify the file in the next line
#     RESTART_FILE_NAME to_be_specified
    &END BFGS
    MAX_ITER 500 #Maximum number of geometry optimization
    MAX_DR 3E-3 #Maximum geometry change
    RMS_DR 1.5E-3 #RMS geometry change
    MAX_FORCE 4.5E-4 #Maximum force
    RMS_FORCE 3E-4 #RMS force
  &END GEO_OPT
  &CONSTRAINT
    &FIXED_ATOMS #Set atoms to be fixed
      COMPONENTS_TO_FIX XYZ #Which fractional components will be fixed, can be X, Y, Z, XY, XZ, YZ, XYZ
      LIST       1..112
    &END FIXED_ATOMS
  &END CONSTRAINT
  &PRINT
    &TRAJECTORY
      FORMAT xyz
    &END TRAJECTORY
    &RESTART
      BACKUP_COPIES 0 #Maximum number of backing up restart file, 0 means never
    &END RESTART
  &END PRINT
&END MOTION

**Comment by fstein93** [comment]
You should better ask these kinds of questions to the user group. The actual output may also help. Does the calculation converge at all (but just slowly)? My guess is a too low cutoff. Set the `CUTOFF` value to at least 600. A reasonable minimal value is given by the product of `REL_CUTOFF` (55) and the largest exponent of your basis functions (12 from Oxygen) giving a minimal value of 600-700. Before you start a geometry optimization, you should preoptimize all parameters (usually `EPS_SCF`, `CUTOFF`, `EPS_DEFAULT`, amount of vacuum in your cell, ...) until your quantities of interest (energies, nuclear forces, ...) are converged. In your case, this is done by using the cheaper `RUN_TYPE ENERGY_FORCE` instead of `RUN_TYPE GEO_OPT`.

**Comment by RMeli** [comment]
> You should better ask these kinds of questions to the user group.

GitHub Discussions are enabled for CP2K. Maybe it's worth migrating this sort of questions there? It should be a simple click away (it closes the issue, and opens a GitHub Discussion).
